File: backend/src/main/java/com/ats/service/UserJobPreferenceService.java
package com.ats.service;

import com.ats.dto.UserJobPreferenceDTO;

import java.util.List;

public interface UserJobPreferenceService {

    /**
     * Save user job preferences from career portal connect feature
     */
    UserJobPreferenceDTO savePreference(UserJobPreferenceDTO preferenceDTO);

    /**
     * Get preference by ID
     */
    UserJobPreferenceDTO getPreferenceById(Long id);

    /**
     * Get all preferences for an email
     */
    List<UserJobPreferenceDTO> getPreferencesByEmail(String email);

    /**
     * Get all preferences (for admin)
     */
    List<UserJobPreferenceDTO> getAllPreferences();

    /**
     * Delete preference by ID
     */
    void deletePreference(Long id);

    /**
     * Check if email already has preferences
     */
    boolean emailExists(String email);
}

File: frontend-career/src/services/userJobPreferenceService.ts
import api from "./api";

export interface UserJobPreferenceDTO {
  id?: number;
  email: string;
  categoryIds: number[];
  consentAccepted: boolean;
  consentAcceptedAt?: string;
  categoryNames?: string[];
  createdAt?: string;
}

export const userJobPreferenceService = {
  /**
   * Save user job preferences
   */
  savePreference: async (
    preference: UserJobPreferenceDTO
  ): Promise<UserJobPreferenceDTO> => {
    const response = await api.post("/user-job-preferences", preference);
    return response.data;
  },

  /**
   * Check if email already exists
   */
  checkEmailExists: async (email: string): Promise<boolean> => {
    const response = await api.get("/user-job-preferences/check-email", {
      params: { email },
    });
    return response.data;
  },
};

export default userJobPreferenceService;

